src/linux/frida-helper-backend-glue.c

```diff
@@ -2843,23 +2843,79 @@ frida_find_libc_base (pid_t pid)
 }
 
 #ifdef HAVE_ANDROID
+static gboolean
+frida_find_library_path (pid_t pid, GumAddress base, gchar ** library_path)
+{
+  gboolean result = false;
+  gchar * maps_path;
+  FILE * fp;
+  const guint line_size = 1024 + PATH_MAX;
+  gchar * line, * path;
+
+  *library_path = NULL;
+
+  maps_path = g_strdup_printf ("/proc/%d/maps", pid);
+
+  fp = fopen (maps_path, "r");
+  g_assert (fp != NULL);
+
+  g_free (maps_path);
+
+  line = g_malloc (line_size);
+  path = g_malloc (PATH_MAX);
+
+  while (result == false && fgets (line, line_size, fp) != NULL)
+  {
+    GumAddress start;
+    gint n;
+
+    n = sscanf (line, "%" G_GINT64_MODIFIER "x-%*x %*s %*x %*s %*s %s", &start, path);
+    if (n == 1)
+      continue;
+    g_assert (n == 2);
+
+    if (path[0] == '[')
+      continue;
+
+    if (start == base)
+    {
+      result = true;
+      if (library_path != NULL)
+        *library_path = g_strdup (path);
+    }
+  }
+
+  g_free (path);
+  g_free (line);
+
+  fclose (fp);
+
+  return result;
+}
```

Comment:
Would be good to move this function after `frida_find_library_base()`, forward-declared with `G_GNUC_UNUSED`:

```C
G_GNUC_UNUSED static gboolean frida_find_library_path (pid_t pid, GumAddress base, gchar ** library_path);
```

Reply:
So you don't want it to be guarded by an `#ifdef HAVE_ANDROID` ?

Reply:
Indeed, it feels like it belongs next to `frida_find_library_base()`. Just need a `G_GNUC_UNUSED` declaration to avoid a compiler warning on non-Android. (The linker will eliminate the unused code anyway.)